tools/pipelines/templates/build-docker-service.yml

```diff
@@ -293,16 +293,41 @@ stages:
 
       # Pack
       - ${{ if eq(parameters.pack, true) }}:
+        # Need to create folders outside the docker container; if it creates them, the permissions will be wrong and
+        # we won't be able to write to them from outside the container.
+        - task: Bash@3
+          displayName: Create folder for pack artifact
+          inputs:
+            targetType: 'inline'
+            script: |
+              mkdir -p $(Build.ArtifactStagingDirectory)/pack
+
         - task: Docker@0
           displayName: Pack
           inputs:
             action: 'Run a Docker command'
-            customCommand: 'run -v $(System.DefaultWorkingDirectory)/pack/scoped:/usr/src/pack -t $(baseContainerTag) pnpm lerna exec --no-private -- mv `npm pack` /usr/src/pack'
+            customCommand: 'run -v $(Build.ArtifactStagingDirectory)/pack/scoped:/usr/src/pack -t $(baseContainerTag) pnpm lerna exec --no-private -- mv `npm pack` /usr/src/pack'
+
+        # Need to generate package lists manually, because we don't have access to the scripts/pack-packages.sh
+        # script that build-npm-package.yml uses.
```

Comment:
This isn't quite true. We now have a way to reference any path in the repo within a Dockerfile. I'll dig up an example.

Reply:
Here's the example: https://github.com/microsoft/FluidFramework/blob/58eebc1ca77041c66cbc54c82e0a400e47f9533c/server/routerlicious/Dockerfile#L25-L27

the "root" is the root of the repo. You can use this to copy anything in the repo into the image for use. I discovered the "build context" capability [somewhat recently](https://github.com/microsoft/FluidFramework/pull/17605).

Anyway, this should let you keep more of the build logic in the Dockerfile.

Reply:
Giving it a shot